tests/scripts/ci_perf.py

```diff
@@ -0,0 +1,177 @@
+#ambient jenkins properties
+parser.add_argument('-workspace', dest='workspace', default=os.path.abspath(os.getcwd()))
+parser.add_argument('-gitBranchWithoutOrigin', dest='gitBranchWithoutOrigin', default='FAKE_BRANCHNAME')
+parser.add_argument('-gitBranch', dest='gitBranch', default='origin/FAKE_BRANCHNAME')
+parser.add_argument('-gitCommit', dest='gitCommit', default='FAKE_GIT_COMMIT')
+
+
+
+##########################################################################
+# Helper Functions
+##########################################################################
+
+def log(message):
+    """ Print logging information
+    Args:
+        message (str): message to be printed
+    """
+
+    print('[%s]: %s' % (sys.argv[0], message))
+
+def is_supported_version() -> bool:
+    """ Checkes that the version of python is at least 3.5
+    Returns:
+        bool : true if python version is 3.5 or higher
+    """
+    return version_info.major > 2 and version_info.minor > 4
+
+def run_command(runArgs, environment):
+    """ Run command specified by runArgs
+    Args:
+        runargs (str[]): list of arguments for subprocess
+        environment(str{}): dict of environment variable
+    """
+    log('RUNNING COMMAND: ' + runArgs)
+    subprocess.check_call(runArgs, env=environment)
+
+##########################################################################
+# CI jobs
+##########################################################################
+
+class jenkins_properties:
+    def __init__(self, workspace, git_branch_without_origin, git_branch, git_commit):
+        self.workspace = workspace
+        self.git_branch_without_origin = git_branch_without_origin
+        self.git_branch = git_branch
+        self.git_commit = git_commit
+
+
+class coreclr_scenarios_args:
+    def __init__(self, is_pr, arch, os, jit_name, opt_level, benchview_commit_name, bv_upload_sas_token, upload_to_benchview):
+        self.is_pr = is_pr
+        self.arch = arch
+        self.os = os
+        self.jit_name = jit_name
+        self.opt_level = opt_level
+        self.benchview_commit_name = benchview_commit_name
+        self.bv_upload_sas_token = bv_upload_sas_token
+        self.upload_to_benchview = upload_to_benchview
+
+def coreclr_scenarios(args, props):
+    architecture = args.arch
+    testEnv = ''
+    configuration = 'Release'
+    runType = 'private' if args.is_pr else 'rolling'
+    benchViewName = 'CoreCLR-Scenarios private ' + args.benchview_commit_name if args.is_pr else 'CoreCLR-Scenarios rolling ' + props.git_branch_without_origin + ' ' + props.git_commit
+    uploadString = '-uploadToBenchview' if args.upload_to_benchview else ''
+    
+    run_command('powershell -NoProfile wget https://dist.nuget.org/win-x86-commandline/latest/nuget.exe -OutFile \"' + props.workspace + '\\nuget.exe\"', os.environ)
+    if os.path.isdir(props.workspace + '\\Microsoft.BenchView.JSONFormat'):
+        shutil.rmtree(props.workspace + '\\Microsoft.BenchView.JSONFormat')
```

Comment:
Do you need to remove it? Wouldn't it make more sense to just use the Benchview tools you already have downloaded if they already exist? These tools rarely change, so you'd be pretty safe only downloading them in script if the directory doesn't exist (and if you're seeing an error in the lab that you don't see locally, you could delete the directory manually).

Reply:
When building a path, please use `os.path.join`

Reply:
instead of `%WORKSPACE%\Microsoft.BenchView.JSONFormat`, can we change it to: `%WORKSPACE%\bin\Microsoft.BenchView.JSONFormat`?

Reply:
I agree that all of these suggestions would be improvements to the algorithm, but my priority in this script was to repro issues that might occur in CI, and to do that I tried to stay faithful to the source material.  All of the issues you call out are present in perf.groovy. I'd be glad to see perf.groovy improved as you suggest, but that is beyond the scope of work I'm willing to undertake.

If you want to do things you are suggesting I'd suggest someone else could either:
a) update perf.groovy to call this script directly and then improve this script as desired.
b) update perf.groovy with these improvements, then update this script to match